mm2src/coins/z_coin.rs

```diff
@@ -887,7 +893,7 @@ impl SwapOps for ZCoin {
                 script_data,
                 key_pair.private().secret.as_slice(),
             );
-            let tx = try_s!(tx_fut.await);
+            let tx = try_tx_s!(tx_fut.await);
```

Comment:
Sorry, I didn't understand what exactly you mean here, it's already returning `TransactionFutErr`.

Reply:
I meant that `z_p2sh_spend` currently returns `ZP2SHSpendError`, but it could return an error extended by the `TxRecoverable` variant or something like that. It actually can return here
https://github.com/KomodoPlatform/atomicDEX-API/blob/5ba0dd1684ce879b4724c6c9badcbe524537f228/mm2src/coins/z_coin/z_htlc.rs#L164-L167
So maybe we can add `ZP2SHSpendError::TxRecoverable` 🤔

Reply:
> I meant that `z_p2sh_spend` currently returns `ZP2SHSpendError`, but it could return an error extended by the `TxRecoverable` variant or something like that. It actually can return here
> 
> https://github.com/KomodoPlatform/atomicDEX-API/blob/5ba0dd1684ce879b4724c6c9badcbe524537f228/mm2src/coins/z_coin/z_htlc.rs#L164-L167
> 
> 
> So maybe we can add `ZP2SHSpendError::TxRecoverable`

Implemented it now.

@artemii235 I can rollback the commit if this isn't necessary.